qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/page/AdministrationManageRolesPage.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ * 
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.contrib.qaframework.page;
+
+import org.openmrs.contrib.qaframework.helper.Page;
+import org.openqa.selenium.By;
+
+public class AdministrationManageRolesPage extends Page{
+
+	protected static final String PAGE_URL = "/openmrs/admin/users/role.list";
+	private static final By ADD_NEW_ROLE_LINK= By.cssSelector("#content > a");
+	private static final By DELETE_SELECTED_ROLES_BUTTON = By.cssSelector("#content > form > input[type=submit]");
+	private static final By CHECKBOX = By.cssSelector("#content > form > table > tbody input[type=checkbox]");
+	private static final By EDIT_ROLE_LINK = By.cssSelector("#content > form > table > tbody > tr:nth-child(3) > td:nth-child(2) > a");
```

Comment:
use a better element reference to EDIT_ROLE_LINK. Retrieving page elements via x-child value get broken when UI changes comes in.

Reply:
i have a challenge selecting a role to edit (without the n-th child locator since i hardwired it before) and now that i remove the nth child, the locators are ambiguous hence the element i want can't be retrieved thus the test fails

Reply:
@kdaud thanks for the reviews,  definitely am gonna improve my commit messages 😆. 

Please take a look at the comment [above](https://github.com/openmrs/openmrs-contrib-qaframework/pull/270#discussion_r770549896)